### PR TITLE
No longer detect .spec.js files

### PR DESCRIPTION
**Pull request by acostalima**

### CODE REVIEW

**Comment by rafaelramalho19** [comment]
This is a breaking change, please use the conventional changelog commits to refer that in the commit message.

**Comment by acostalima** [comment]
> This is a breaking change, please use the conventional changelog commits to refer that in the commit message.

@satazor 👆

**Comment by satazor** [comment]
There’s no project using .spec.files that is using v2 of this package. While technically this is a breaking change, we could skip signaling it since we know it won’t cause issues for us. 

If you have a strong opinion on signaling this as a breaking change, I won’t opposite to it.

File: lib/base-config.js [hunk 1]
```diff
@@ -28,4 +28,8 @@ module.exports = () => ({
     snapshotSerializers: [
         require.resolve('jest-serializer-path'),
     ],
+    testMatch: [
+        '**/__tests__/**/*.[jt]s?(x)',
+        '**/?(*.)test.[jt]s?(x)',
+    ],
```
Comment (acostalima) [inline lib/base-config.js]: @satazor I'm now wondering what's the point of having the matching pattern `'**/__tests__/**/*.[jt]s?(x)'`. Doesn't `'**/?(*.)test.[jt]s?(x)'` pretty much cover all the scenarios we're looking for?

**Comment by acostalima** [comment]
> There’s no project using .spec.files that is using v2 of this package. While technically this is a breaking change, we could skip signaling it since we know it won’t cause issues for us.
> 
> If you have a strong opinion on signaling this as a breaking change, I won’t oppose to it.

I'm fine with this as well.

File: lib/base-config.js (hunk 1 above)
Comment (acostalima) [reply lib/base-config.js]: If I have co-localized utils to support my tests in the `__tests__` folder, Jest will still detect them.

Comment (satazor) [reply lib/base-config.js]: That pattern allow us to have tests files with the suffix in test folders. That’s pattern is Jest defaults.

Comment (satazor) [reply lib/base-config.js]: Fixed my comment.

Comment (satazor) [reply lib/base-config.js]: If we want to enforce all test files to end with `.test.js`, even if they are already in `__tests__` or `test`, then we may tweak the pattern.

I don’t have a strong opinion on this one, however maybe we should keep the pattern close to Jest defaults? (Disallowing just spec ones)

Comment (satazor) [reply lib/base-config.js]: > If I have co-localized utils to support my tests in the `__tests__` folder, Jest will still detect them. We don't want this, right?

That’s a good point to actually restrict test files to `.test.js`.

Comment (acostalima) [reply lib/base-config.js]: I've just dropped `'**/__tests__/**/*.[jt]s?(x)'` pattern.

**Comment by acostalima** [comment]
> > There’s no project using .spec.files that is using v2 of this package. While technically this is a breaking change, we could skip signaling it since we know it won’t cause issues for us.
> > If you have a strong opinion on signaling this as a breaking change, I won’t oppose to it.
> 
> I'm fine with this as well.

Breaking changed introduced in the commit message.

File: lib/base-config.js (hunk 1 above)
Comment (acostalima) [reply lib/base-config.js]: @moxystudio/developers, other opinions are welcomed. 💪

File: README.md [hunk 2]
```diff
@@ -42,6 +42,8 @@ module.exports = baseConfig();
 - **Coverage thresholds:** For a good balance  between strict but workable thresholds.
 - **Snapshot serializing:** To remove absolute paths from your snapshots, reducing conflicts in CI.
 
+ℹ️ Test files ending in .spec.js are no longer detected.
```
Comment (satazor) [inline README.md]: We might want to change this message to something like:

`ℹ️ Test files must be named `test.js` or end with `.test.js` , even if they are inside `__tests__` or any other folder.

File: lib/base-config.js (hunk 1 above)
Comment (zebateira) [reply lib/base-config.js]: > I've just dropped '**/__tests__/**/*.[jt]s?(x)' pattern.

agree 👍

File: README.md (hunk 2 above)
Comment (acostalima) [reply README.md]: Done! ✅

**Pull request merged**